File: FixMyCar/FixMyCar.Model/Entities/Reservation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FixMyCar.Model.Entities
{
    public class Reservation
    {
        public int Id { get; set; }
        [ForeignKey(nameof(CarRepairShop))]
        public int CarRepairShopId { get; set; }
        public CarRepairShop CarRepairShop { get; set; }
        [ForeignKey(nameof(Client))]
        public int ClientId { get; set; }
        public Client Client { get; set; }
        [ForeignKey(nameof(Order))]
        public int? OrderId { get; set; }
        public Order? Order { get; set; }
        public bool? ClientOrder { get; set; }
        public DateTime ReservationCreatedDate { get; set; }
        public DateTime ReservationDate { get; set; }
        public DateTime? EstimatedCompletionDate { get; set; }
        public DateTime? CompletionDate { get; set; }
        public double TotalAmount { get; set; }
        public TimeSpan TotalDuration { get; set; }
        [ForeignKey(nameof(CarRepairShopDiscount))]
        public int? CarRepairShopDiscountId { get; set; }
        public CarRepairShopDiscount? CarRepairShopDiscount { get; set; }
        public string State { get; set; }
        public string Type { get; set; }
        public string PaymentMethod { get; set; }
        public ICollection<ReservationDetail> ReservationDetails { get; set; }
    }
}

File: FixMyCar/FixMyCar.UI/fixmycar_client/lib/src/models/reservation/reservation.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int id;
  int carRepairShopId;
  String carRepairShopName;
  String clientUsername;
  int? orderId;
  bool? clientOrder;
  String reservationCreatedDate;
  String reservationDate;
  String? estimatedCompletionDate;
  String? completionDate;
  double totalAmount;
  String totalDuration;
  double carRepairShopDiscountValue;
  String state;
  String type;

  Reservation(
      this.id,
      this.carRepairShopId,
      this.carRepairShopName,
      this.clientUsername,
      this.orderId,
      this.clientOrder,
      this.reservationCreatedDate,
      this.reservationDate,
      this.estimatedCompletionDate,
      this.completionDate,
      this.totalAmount,
      this.totalDuration,
      this.carRepairShopDiscountValue,
      this.state,
      this.type);

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
